ckanapi/common.py

```diff
@@ -61,7 +62,24 @@ def is_file_like(v):
     return hasattr(v, 'read') or (
         isinstance(v, tuple) and len(v) >= 2 and hasattr(v[1], 'read'))
 
+def _encodehack(s):
```

Comment:
The problem here was that `str.encode()` produces an object of type `bytes` in Python3. This leads to failure as further downstream the data is expected to be of type `str`. I tried to fix this without changing behaviour in Python2 but I believe the code did't work as intended in the first place. Here is how I believe it is supposed to work and how it works now:

In Python2:
+ convert `unicode` strings to UTF-8 encoded byte strings.
+ leave a byte string unchanged if it is a valid UTF8-encoding. Otherwise raise an exception.

In Python3:
+ convert everything to a (unicode) `str` that can be converted  by assuming UTF-8 encoding.
+ raise an exception if given a non-decodable `bytes`string.